src/main/java/nextstep/app/oauth2/GithubOAuth2UserDetailService.java

```diff
@@ -0,0 +1,49 @@
+package nextstep.app.oauth2;
+
+import nextstep.security.oauth2.userdetails.OAuth2UserDetails;
+import nextstep.security.oauth2.userdetails.OAuth2UserDetailsService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Map;
+import java.util.Set;
+
+@Component
+public class GithubOAuth2UserDetailService implements OAuth2UserDetailsService {
+
+    private final RestTemplate restTemplate;
+    @Value("${app.oauth2.github.user-url}")
+    private String userUrl;
+
+    public GithubOAuth2UserDetailService(final RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
```

Comment:
```suggestion
    private final RestTemplate restTemplate;
    private final String userUrl;

    public GithubOAuth2UserDetailService(
            final RestTemplate restTemplate,
            @Value("${app.oauth2.github.user-url}") final String userUrl
    ) {
        this.restTemplate = restTemplate;
        this.userUrl = userUrl;
    }
```

통일성있게 userUrl도 final로 만들어보면 좋겠네요!

Reply:
오 이 부분은 코틀린에서도 되는건데 자바에서도 된다는 생각을 못했네요! 감사합니다!!